File: dotnet/Allmark/Render/ListTaskItemRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ListTaskItemRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "list_task_item",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        bool isChecked = node.Markup.Length > 1 && !Utils.IsSpace(node.Markup[1]);
        string checkedAttr = isChecked ? " checked=\"\"" : "";
        state.Output.Append($"<input type=\"checkbox\"{checkedAttr} disabled=\"\" /> ");
    }
}

File: web/src/render-console/listTaskItemRenderer.ts
import type ConsoleRendererState from "../types/ConsoleRendererState";
import type MarkdownNode from "../types/MarkdownNode";
import type Renderer from "../types/Renderer";
import type RendererState from "../types/RendererState";
import ANSI from "./ansi";

const renderer: Renderer = {
	name: "list_task_item",
	render,
};
export default renderer;

function render(node: MarkdownNode, state: RendererState): void {
	const s = state as ConsoleRendererState;
	const style = ANSI.dim;
	const reset = ANSI.reset;

	const isChecked = node.markup?.[1] !== " ";
	const emoji = isChecked ? "✓" : " ";
	s.output += `${style}[${reset}${emoji}${style}]${reset} `;
}
